app_dart/lib/src/request_handlers/create_agent.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:cocoon_service/src/model/appengine/agent.dart';
+import 'package:cocoon_service/src/request_handling/exceptions.dart';
+import 'package:gcloud/db.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+import '../service/agent_service.dart';
+import '../service/datastore.dart';
+
+@immutable
+class CreateAgent extends ApiRequestHandler<CreateAgentResponse> {
+  const CreateAgent(
+    Config config,
+    AuthenticationProvider authenticationProvider, {
+    @visibleForTesting
+        this.datastoreProvider = DatastoreService.defaultProvider,
+    this.agentServiceProvider = AgentService.defaultProvider,
+  }) : super(config: config, authenticationProvider: authenticationProvider);
+
+  final DatastoreServiceProvider datastoreProvider;
+  final AgentServiceProvider agentServiceProvider;
+
+  static const String agentIdParam = 'AgentID';
+  static const String capabilitiesParam = 'Capabilities';
+
+  @override
+  Future<CreateAgentResponse> post() async {
+    checkRequiredParameters(<String>[agentIdParam, capabilitiesParam]);
+
+    final String agentId = requestData[agentIdParam];
+    final List<String> capabilities =
+        requestData[capabilitiesParam].cast<String>().toList();
+    final DatastoreService datastore = datastoreProvider();
+    final AgentService agentService = agentServiceProvider();
+    final Key key = datastore.db.emptyKey.append(Agent, id: agentId);
+
+    Agent agent;
+    AgentAuthToken agentAuthToken;
+
+    if (await datastore.db.lookupValue<Agent>(key, orElse: () => null) ==
```

Comment:
Nit: if you structured this as `if (...) { throw ... }`, then you could set the `agent` and `agentAuthToken` variables when you declare them.

Reply:
You are right! Just updated accordingly.